Extend/Model/Payment_Gateway.php

```diff
@@ -168,35 +119,41 @@ public function getRedirectUrls(Session $oSession, string $sShopUrl): Redirect
             $sSid = '&' . $sSid;
         }
 
-        $sErrorText = Helper::translate('order_error');
+        $sRandom = substr(str_shuffle(md5(time())), 0, 15);
+        $oSession->setVariable("wdtoken", $sRandom);
+
+        $sBaseLanguage = Registry::getLang()->getBaseLanguage();
+
         $oRedirect = new Redirect(
-            $sShopUrl . 'index.php?cl=thankyou' . $sSid,
-            $sShopUrl . 'index.php?type=cancel&cl=payment',
-            $sShopUrl . 'index.php?type=error&cl=payment&errortext=' . urlencode($sErrorText)
+            $sShopUrl . 'index.php?lang=' . $sBaseLanguage . '&cl=order&wdpayment=' . $sRandom . $sSid,
+            $sShopUrl . 'index.php?lang=' . $sBaseLanguage . '&cl=payment&payerror=-100',
+            $sShopUrl . 'index.php?lang=' . $sBaseLanguage . '&cl=payment&payerror=-101'
```

Comment:
Can you add the ssid to these two redirects as well!? Just to be sure to be in the right session on the redirect calls